src/ORM/TableRegistry.php

```diff
@@ -184,7 +184,7 @@ public static function get($alias, array $options = [])
         }
 
         if (isset(static::$_config[$alias])) {
-            $options += static::$_config[$alias];
+            $options = static::$_config[$alias] + $options;
```

Comment:
@dereuromark The earlier code was not properly merging the `className` option set via `TableRegistry::config()`.